R/io.R

```diff
@@ -0,0 +1,131 @@
+#' Manage a gitsum repository
+#'
+#' A gitsum repository is a git repository with an additional .gitsum folder in
+#' the root directory that is mainly used to store a parsed git log as a tibble.
+#' @param path The path to the repository to initialize.
+#' @param over_write Whether or not an existing directory should be overwritten.
+#' @name manage_gitsum
+#' @aliases init_gitsum remove_gitsum
+NULL
+
+#' @describeIn manage_gitsum Calling this function will parse the available git
+#'   history and dump it into the directory .gitsum.
+#' @export
+init_gitsum <- function(path = ".", over_write = FALSE) {
+  check_overwriting_clearance(gitsum_path(path), over_write, dir.exists)
+  parse_log_detailed_full_run(path) %>%
+    update_dump_from_log(path)
+}
+
+#' @describeIn manage_gitsum Removes the gitsum repository, i.e just the folder
+#' .gitsum and its contents.
+#' @export
+remove_gitsum <- function(path = ".") {
```

Comment:
Not sure it really matters in this case, but wouldn't it make more sense for this to be something along the lines of:
```
was_gitsum_repo <- is_gitsum_repo(path)
if (was_gitsum_repo) {
  unlink(gitsum_path(path), recursive = TRUE)
  message(".gitsum removed.")
}
```

Reply:
Right, we could do that.